pytensor/tensor/math.py

```diff
@@ -343,6 +345,208 @@ def grad(self, inp, grads):
+        # reshaped_x = transposed_x.reshape(new_shape)
+
+        # max_idx[0] = _asarray(np.argmax(reshaped_x, axis=-1), dtype="int64")
+
+    def c_code(self, node, name, inp, out, sub):
+        if len(self.axis) != 1 and len(self.axis) != node.inputs[0].ndim:
+            raise NotImplementedError(
+                "NumPy C-API can compute max only for 1 axis or for all axes."
+            )
+        x = inp[0]
+        axis = sub["params"]
+        # max, argmax = out
+        (max,) = out
+        fail = sub["fail"]
+        ret = """
+        #if PY_MAJOR_VERSION >= 3
+            #ifndef PyInt_AS_LONG
+                #define PyInt_AS_LONG PyLong_AS_LONG
+            #endif
+        #endif
+
+        int axis;
+
+        if (PyTuple_GET_SIZE(%(axis)s) == PyArray_NDIM(%(x)s)) {
+            axis = NPY_MAXDIMS;
+        } else if(PyTuple_GET_SIZE(%(axis)s) == 1) {
+            PyObject* axis_object = PyTuple_GET_ITEM(%(axis)s, 0);
+            axis = (int)PyInt_AS_LONG(axis_object);
+            if (axis > PyArray_NDIM(%(x)s)-1 || axis < -PyArray_NDIM(%(x)s)) {
+                PyErr_SetString(PyExc_ValueError,
+                "TensorMax: bad axis argument");
+                %(fail)s
+            }
+        } else {
+            PyErr_SetString(PyExc_NotImplementedError,
+            "TensorMax: NumPy C-API can compute max only for 1 axis or for all axes.");
+            %(fail)s
+        }
+
+        Py_CLEAR(%(max)s);
+
+        %(max)s = (PyArrayObject*)PyArray_Max(%(x)s, axis, NULL);
+        if (%(max)s == NULL) {
+            %(fail)s;
+        }
+        if (!PyArray_CheckExact(%(max)s)) {
+            %(max)s = (PyArrayObject*)PyArray_FromAny((PyObject*)%(max)s, NULL, 0, 0, NPY_ARRAY_ENSUREARRAY, NULL);
+            if(%(max)s == NULL){
+                %(fail)s;
+            }
+        }
+        """
+        return ret % locals()
+
+    def c_code_cache_version(self):
+        return (5,)
+
+    def infer_shape(self, fgraph, node, shapes):
+        ishape = shapes[0]
+        rval = tuple(
+            ishape[i]
+            for (i, b) in enumerate(node.inputs[0].type.broadcastable)
+            if i not in self.axis
+        )
+        return [rval]
+
+    def R_op(self, inputs, eval_points):
+        if eval_points[0] is None:
+            return [None, None]
+
+        if len(self.axis) != 1:
+            raise ValueError("R_op supported for arg_max only for one axis!")
+        if self.axis[0] > 1:
+            raise ValueError("R_op supported for arg_max only when axis is 0 or 1")
+        if inputs[0].ndim != 2:
+            raise ValueError("R_op supported for arg_max only when input is a matrix")
+        # max_vals, max_pos = self.make_node(*inputs).outputs
+        # max_vals = self.make_node(*inputs).outputs
+        if self.axis[0] == 0:
+            return [eval_points[0][arange(eval_points[0].shape[1])], None]
+        else:
+            return [eval_points[0][arange(eval_points[0].shape[0])], None]
```

Comment:
Need help regarding the ```R_op``` function. Do not know if this implementation is the correct usage or not.